login-workflow/src/new-architecture/screens/ForgotPasswordScreen/ForgotPasswordScreen.tsx

```diff
@@ -21,15 +21,18 @@ export const ForgotPasswordScreen: React.FC<ForgotPasswordScreenProps> = (props)
 
     const EMAIL_REGEX = /^[A-Z0-9._%+'-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i;
 
-    const handleOnNext = useCallback(async (): Promise<void> => {
-        try {
-            setIsLoading(true);
-            await actions().forgotPassword(emailInput);
-            setIsLoading(false);
-        } catch (e) {
-            setShowErrorDialog(true);
-        }
-    }, [setIsLoading, setShowErrorDialog, actions, emailInput]);
+    const handleOnNext = useCallback(
+        async (email: string): Promise<void> => {
+            try {
+                setIsLoading(true);
+                await actions().forgotPassword(email);
+                setIsLoading(false);
+            } catch (e) {
+                setShowErrorDialog(true);
```

Comment:
It will first set the loader and call async action:
success:
 Will set the loader to false
 Do the rest of the thing (mostly navigate to next page)
 
 failure(catch):
 Will set the error dialog